### PR TITLE
aichat: update page context toggle design

### PR DESCRIPTION
**Pull request by nullhook**
Resolves https://github.com/brave/brave-browser/issues/38127
Resolves https://github.com/brave/brave-browser/issues/36112
Resolves https://github.com/brave/brave-browser/issues/35989

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. hover or tap should show tooltip
2. tap radius should be improved on mobile
3. tap info icon should show tooltip and tapping outside should make it dissapear

### CODE REVIEW

File: components/ai_chat/resources/page/components/page_context_toggle/style.module.scss [hunk 1]
```diff
@@ -48,6 +45,20 @@
   padding: var(--leo-spacing-m);
 }
 
+.tooltipButton {
+  --leo-button-padding: 0;
+  width: 24px;
```
Comment (fallaciousreasoning) [inline components/ai_chat/resources/page/components/page_context_toggle/style.module.scss:50]: Out of interest can you not use a `fab` button?

`<Button fab kind="plain-faint" size="tiny">`

You'll probably still need to set the icon color.

**Review by fallaciousreasoning — APPROVED** [review]

File: components/ai_chat/resources/page/components/page_context_toggle/style.module.scss (hunk 1 above)
Comment (nullhook) [reply components/ai_chat/resources/page/components/page_context_toggle/style.module.scss:50]: does fab eliminate the need setting the button padding, is that all?

Comment (fallaciousreasoning) [reply components/ai_chat/resources/page/components/page_context_toggle/style.module.scss:49]: It also make the button have a 1:1 aspect ratio

File: components/ai_chat/resources/page/components/page_context_toggle/index.tsx [hunk 2]
```diff
@@ -54,16 +47,14 @@ function PageContextToggle() {
...
               </div>
             </div>
-            <div
-              onMouseOver={() => setShowTooltip(true)}
-              onMouseOut={() => setShowTooltip(false)}
-              onTouchStart={handleInfoIconTouchStart}
-              onTouchEnd={handleInfoIconTouchEnd}
-              onFocus={() => setShowTooltip(true)}
-              onBlur={() => setShowTooltip(false)}
+            <Button
```
Comment (fallaciousreasoning) [inline components/ai_chat/resources/page/components/page_context_toggle/index.tsx:50]: Sorry to nitpick but I'm surprised the `preventDefault()` works on the Button - it shouldn't be an HTML event, but instead the wrapped Svelte event

**Review by fallaciousreasoning — APPROVED** [review]

File: components/ai_chat/resources/page/components/page_context_toggle/index.tsx (hunk 2 above)
Comment (nullhook) [reply components/ai_chat/resources/page/components/page_context_toggle/index.tsx:65]: ok, so i have made the tooltip controlled which eliminates the need for preventDefault() and has a better UX as tapping the info icon should toggle visibility on mobile

**Comment by SergeyZhukovsky** [comment]
I checked it on Android and it works as expected:

[Screen_recording_20240508_131022.webm](https://github.com/brave/brave-core/assets/12011303/050ccb6f-7395-4cf6-97ef-ae510ab467a7)

**Review by fallaciousreasoning — APPROVED** [review]

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**